.circleci/test-deploy.yml

```
@@ -0,0 +1,143 @@
+version: 2.1
+orbs:
+  # Your orb will be automatically injected here during the pipeline.
+  # Reference your orb's jobs and commands below as they will exist when built.
+  orb-tools: circleci/orb-tools@12.1
+  # The orb definition is intentionally not included here. It will be injected into the pipeline.
+  cloudsmith-maven: {}
+
+# Use this tag to ensure test jobs always run,
+# even though the downstream publish job will only run on release tags.
+filters: &filters
+  tags:
+    only: /.*/
+
+# Filter for release tags.
+release-filters: &release-filters
+  branches:
+    ignore: /.*/
+  tags:
+    only: /^v[0-9]+\.[0-9]+\.[0-9]+$/
+
+jobs:
+
+  test-set_env_vars_for_maven:
+    docker:
+      - image: cimg/openjdk:21.0.2
+    resource_class: small
+    steps:
+      - checkout
+      - cloudsmith-maven/set_env_vars_for_maven:
+          repository: "circleci-orb-testing"
+      - run:
+          name: Check that maven environment variables have been set
+          command: |
+            if [ -z $CLOUDSMITH_MAVEN_REPOSITORY_URL ]
+            then
+              echo "Test failed: CLOUDSMITH_MAVEN_REPOSITORY_URL has not been set."
+              exit 1
+            fi
+            if [ -z "$CLOUDSMITH_REPOSITORY" ]
+            then
+              echo "Unable to configure MAVEN. Env var CLOUDSMITH_REPOSITORY is not defined. Please run the set_env_vars_for_maven command first."
+              exit 1
+            fi
+            if [ -z "$CLOUDSMITH_OIDC_TOKEN" ]
+            then
+              echo "Unable to configure MAVEN. Env var CLOUDSMITH_OIDC_TOKEN is not defined. Please run the set_env_vars_for_maven command first."
+              exit 1
+            fi
+            echo "All tests passed."
+
+  test-configure_maven_install_package:
+    docker:
+      - image: cimg/openjdk:21.0.2
+    resource_class: small
+    steps:
+      - checkout
+      - cloudsmith-maven/configure_maven:
+          repository: "circleci-orb-testing"
+      - run:
+          name: Check that maven environment variables have been set
+          command: |
+            if [ -z $CLOUDSMITH_MAVEN_REPOSITORY_URL ]
+            then
+              echo "Test failed: CLOUDSMITH_MAVEN_REPOSITORY_URL has not been set."
+              exit 1
+            fi
+            if [ -z "$CLOUDSMITH_REPOSITORY" ]
+            then
+              echo "Unable to configure MAVEN. Env var CLOUDSMITH_REPOSITORY is not defined. Please run the set_env_vars_for_maven command first."
+              exit 1
+            fi
+            if [ -z "$CLOUDSMITH_MAVEN_TOKEN" ]
+            then
+              echo "Unable to configure MAVEN. Env var CLOUDSMITH_MAVEN_TOKEN is not defined. Please run the set_env_vars_for_maven command first."
+              exit 1
+            fi
+            echo "All tests passed."
+      - run:
+          name: cd into maven project directory and install package dependencies
```

```suggestion
          name: Install maven package dependencies
```
We added the _working_directory_ param, so we no longer need to "cd" into the project directory.